Examples/Python/python/acts/examples/__init__.py

```diff
@@ -285,3 +286,43 @@ def _allmods(mod, base, found):
     if not quiet and donemods:
         print("dump_args for module functions:", ", ".join(donemods))
     return alldone
+
+
+class CustomLogLevel(Protocol):
+    def __call__(
+        self,
+        override: Optional[acts.logging.Level] = None,
+        min: acts.logging.Level = acts.logging.VERBOSE,
+        max: acts.logging.Level = acts.logging.FATAL,
+    ) -> acts.logging.Level:
+        ...
+
+
+def defaultLogging(
+    s=None,
+    logLevel: Optional[acts.logging.Level] = None,
+) -> CustomLogLevel:
+    """
+    Establishes a default logging strategy for the python examples interface.
+
+    Returns a function that determines the log level in the following schema:
+    - if `logLevel` is set use it otherwise use the log level of the sequencer `s.config.logLevel`
+    - the returned log level is bound between `minLevel` and `maxLevel` provided to `customLogLevel`
+
+    Examples:
+    - `customLogLevel(minLevel=acts.logging.INFO)` to get a log level that is INFO or higher
+      (depending on the sequencer and `logLevel` param) which is useful to suppress a component which
+      produces a bunch of logs below INFO and you are actually more interested in another component
+    - `customLogLevel(maxLevel=acts.logging.INFO)` to get a log level that is INFO or lower
+      (depending on the sequencer and `logLevel` param) which is useful to get more details from a
+      component that will produce logs of interest below the default level
+    """
```

Comment:
This is a good explanation, but let me raise a long-standing concern (not specific to your changes).

I've been wondering if we need to be clearer about the potential confusion from *lower* `logLevel` giving *more* output. That's not a problem, until we compare > or < to a `minLevel` or `maxLevel`.
Put another way, the `minLevel` defines the maximum amount of logging; the `maxLevel` defines the minimum amount of logging.
Unless someone can think of a better argument name, I think the best we can do is document it. Do you think it's worth adding a comment to this effect, or maybe the nice new examples are enough?

Reply:
I agree this is quite confusing and I always need to think twice about it :smile: 

I don't see an easy way out of this except better doc as you said.

maybe instead of min max which introduces numbers for comparison with > or < we could rephrase this `minLevel` -> `ignoreBelowLevel` and `maxLevel` -> `requrieAboveLevel`

Reply:
> we could rephrase this `minLevel` -> `ignoreBelowLevel` and `maxLevel` -> `requireAboveLevel`
I want to like this, but to me it is perhaps even more confusing. But don't take my word on it, perhaps it is clearer to others.

Reply:
in that case I would just default to the min/max for now and we can improve it at any point

Reply:
yeah. Maybe add a sentence to the docs explaining this?